src/test/java/com/wikia/webdriver/elements/mercury/components/RecentWikiActivity.java

```diff
@@ -0,0 +1,148 @@
+
+    return this;
+  }
+
+  public RecentWikiActivity openDiffPage() {
+    wait.forElementClickable(seeDiffArrow);
+
+    String urlToBeRedirected = seeDiffArrow.getAttribute("href");
+    seeDiffArrow.click();
+
+    loading.handleAsyncPageReload();
+
+    PageObjectLogging.logInfo("The arrow redirecting to the diff page was clicked");
+
+    Assertion.assertTrue(driver.getCurrentUrl().contains(urlToBeRedirected),
+                         "You were not redirected to the diff page");
+    PageObjectLogging.logInfo("You were redirected to the diff page");
+
+    return this;
+  }
+
+  private void submitOnDiffPage() {
+    wait.forElementClickable(undoButton);
+    PageObjectLogging.logInfo("Undo button is clickable");
+
+    undoButton.click();
+    PageObjectLogging.logInfo("Undo button was clicked");
+  }
+
+  private void summary(String searchQuery) {
+    wait.forElementClickable(summaryTestbox);
+    summaryTestbox.click();
+    actions.sendKeys(searchQuery).perform();
+    PageObjectLogging.logInfo("Typed in summary box: " + searchQuery);
+  }
+
+  private void submitOnSummaryPage() {
+    wait.forElementClickable(undoButtonOnSummaryPage);
+    undoButtonOnSummaryPage.click();
+    PageObjectLogging.logInfo("Undo button was clicked");
+
+    loading.handleAsyncPageReload();
+
+    Assertion.assertTrue(driver.getCurrentUrl().contains(URLsContent.RECENT_WIKI_ACTIVITY),
+                         "You were not redirected to the recent wiki activity");
+    PageObjectLogging.logInfo("You were redirected to the recent wiki activity");
+  }
+
+  public RecentWikiActivity displaySuccessNotification() {
+    wait.forElementVisible(successNote);
+    PageObjectLogging.logInfo("Success notification was displayed");
+
+    return this;
+  }
+
+  public RecentWikiActivity submitWithoutSummary() {
+    submitOnDiffPage();
+    submitOnSummaryPage();
+    PageObjectLogging.logInfo("Undo was submitted without any summary");
+
+    return this;
+  }
+
+  public RecentWikiActivity submitWithSummary() {
+    submitOnDiffPage();
+    summary("For testing");
+    submitOnSummaryPage();
+    PageObjectLogging.logInfo("Undo was submitted with a short summary");
+
+    return this;
+  }
+
+  public RecentWikiActivity goBackFromSummaryPage() {
+    submitOnDiffPage();
+
+    wait.forElementClickable(backButton);
+    backButton.click();
+    PageObjectLogging.logInfo("The X button on the summary page was clicked");
+
+    loading.handleAsyncPageReload();
+
+    wait.forElementVisible(diff);
+    PageObjectLogging.logInfo("You were redirected to the diff page");
+
+    return this;
+  }
+
+  public RecentWikiActivity goBackToRWA() {
+    wait.forElementClickable(backArrow);
+    backArrow.click();
+
+    loading.handleAsyncPageReload();
+
+    PageObjectLogging.logInfo("The arrow that redirect user back to RWA was clicked");
+
+    Assertion.assertTrue(driver.getCurrentUrl().contains(URLsContent.RECENT_WIKI_ACTIVITY),
```

Comment:
Why should i move it to tests? I thought that this assertion is a part of this method so it can stay there and because of that tests look clean.